src/action.ts

```diff
@@ -217,30 +217,38 @@ class Action {
     const logsDir = this.mkdir(path.join(baseDir, "/logs"))
     const reportsDir = this.mkdir(path.join(baseDir, "/reports"))
 
-    const tasksToProcess = executionGraph.tasks
-      .filter(t => t.status === TaskStatus.Succeeded && !this.config.onlyUploadOnFailure || t.status === TaskStatus.Failed)
-
-    for (const task of tasksToProcess) {
+    for (const task of executionGraph.tasks) {
       const taskId = task.task_id
 
+      // API restriction
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      let taskReport: { [key: string]: any } = {}
       try {
-        const logs = await this.vib.getRawLogs(executionGraphId, taskId)
-        const logsFile = this.writeFileSync(path.join(logsDir, `${task.action_id}-${taskId}.log`), logs)
-        core.debug(`Downloaded logs file for task ${taskId}`)
-        artifacts.push(logsFile)
+        taskReport = await this.vib.getTaskReport(executionGraphId, taskId)
```

Comment:
We have to download the task report always. Then, depending whether if it passed or not, and `only-upload-on-failure`, we'll download and upload the artifacts.